cli/src/semgrep/lsp/server.py

```diff
@@ -29,66 +25,23 @@
 # core/osemgrep does't have the target manager yet, we need to do that stuff
 # here.
 class SemgrepCoreLSServer:
-    def __init__(
-        self, rule_file: _TemporaryFileWrapper, target_file: _TemporaryFileWrapper
-    ) -> None:
+    def __init__(self) -> None:
         self.std_reader = JsonRpcStreamReader(sys.stdin.buffer)
         self.std_writer = JsonRpcStreamWriter(sys.stdout.buffer)
-        self.config = LSPConfig({}, [])
-        self.rule_file = rule_file
-        self.target_file = target_file
+        self.config = LSPConfig({})
         self.core_process: Optional[subprocess.Popen] = None
         self.exit_code: int = 0
-
-    def update_targets_file(self) -> None:
-        self.config.refresh_target_manager()
-        self.plan = CoreRunner.plan_core_run(
-            self.config.rules, self.config.target_manager, set()
-        )
-        target_file_contents = json.dumps(self.plan.to_json())
-        # So semgrep-core could try to open the target file when it's empty if
-        # we just truncated then wrote it (since the truncate may write without
-        # a flush call). If we simply write then truncate, it'll be ok because
-        # the core will only take the first occurence of the target json in the
-        # file. Weird
-        self.target_file.seek(0, 0)
-        self.target_file.write(target_file_contents)
-        self.target_file.truncate(len(target_file_contents))
-        self.target_file.flush()
-
-    def update_rules_file(self) -> None:
-        self.config.refresh_rules()
-        rules = self.config.rules
-        rule_file_contents = json.dumps(
-            {"rules": [rule._raw for rule in rules]}, indent=2, sort_keys=True
-        )
-        # Same as before re: sharing files with semgrep-core
-        self.rule_file.seek(0, 0)
-        self.rule_file.write(rule_file_contents)
-        self.rule_file.truncate(len(rule_file_contents))
-        self.rule_file.flush()
+        self.polling: bool = True
 
     def start_ls(self) -> None:
-        cmd = [
-            str(self.config.engine_type.get_binary_path()),
-            "-j",
-            str(self.config.jobs),
-            "-rules",
-            self.rule_file.name,
-            "-targets",
-            self.target_file.name,
-            "-max_memory",
-            str(self.config.max_memory),
-            "-timeout",
-            str(self.config.timeout),
-            "-timeout_threshold",
-            str(self.config.timeout_threshold),
-            "-fast",
-            "-ls",
-        ]
-        if self.config.debug:
-            cmd.append("-debug")
-
+        path = compute_executable_path("osemgrep")
```

Comment:
Ahhhh ok, I was a bit confused about how this worked, I'll clean this up then

Reply:
I think this is the only thing remaining to fix, to call semgrep --experimental instead of osemgrep (which does not exist in the latest distributions).
Otherwise I love the added comments in the .mli Very useful.